apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveService.kt

```diff
@@ -211,20 +170,64 @@ class OppgaveService(
         referanse: String,
     ) {
         val hentetOppgave =
-            oppgaveDao.hentOppgave(oppgaveId)
-                ?: throw IkkeFunnetException(
-                    code = "OPPGAVE_IKKE_FUNNET",
-                    detail = "Oppgaven finnes ikke",
-                    meta = mapOf("oppgaveId" to oppgaveId),
-                )
-        sikreAtOppgaveIkkeErAvsluttet(hentetOppgave)
-        if (hentetOppgave.saksbehandler?.ident.isNullOrEmpty()) {
-            throw OppgaveIkkeTildeltSaksbehandler(oppgaveId)
-        } else {
+            oppgaveDao.hentOppgave(oppgaveId) ?: throw OppgaveIkkeFunnet(oppgaveId)
+        sikreAktivOppgaveOgTildeltSaksbehandler(hentetOppgave) {
             oppgaveDao.endreTilKildeBehandlingOgOppdaterReferanse(oppgaveId, referanse)
         }
     }
 
+    fun endrePaaVent(
+        oppgaveId: UUID,
+        merknad: String,
+        paaVent: Boolean,
+    ) {
+        val oppgave = hentOppgave(oppgaveId) ?: throw OppgaveIkkeFunnet(oppgaveId)
+        if (paaVent && oppgave.status == Status.PAA_VENT) return
+        if (!paaVent && oppgave.status != Status.PAA_VENT) return
+
+        sikreAktivOppgaveOgTildeltSaksbehandler(oppgave) {
+            val nyStatus = if (paaVent) Status.PAA_VENT else Status.UNDER_BEHANDLING
+            oppgaveDao.oppdaterStatusOgMerknad(oppgaveId, merknad, nyStatus)
+            when (oppgave.type) {
+                OppgaveType.FOERSTEGANGSBEHANDLING,
+                OppgaveType.REVURDERING,
+                OppgaveType.ATTESTERING,
+                OppgaveType.UNDERKJENT,
+                -> {
+                    hendelser.sendMeldingForHendelsePaaVent(
+                        UUID.fromString(oppgave.referanse),
+                        if (nyStatus == Status.PAA_VENT) BehandlingHendelseType.PAA_VENT else BehandlingHendelseType.AV_VENT,
+                    )
+                }
+                OppgaveType.TILBAKEKREVING -> {
+                    // TODO
+                }
+                OppgaveType.KLAGE -> {
+                    // TODO
```

Comment:
Hvis statistikkService / repo endres til å håndtere at vi ikke finner noe for behandlingen, så tror jeg vi kan bare sende hendelser på samme måte som for behandling over for både klage og tilbakekreving